Tests/C++WorkshopTests/CardsTests.mm

```diff
@@ -0,0 +1,120 @@
+@import XCTest;
+
+@import C__Workshop;
+
+@interface Cards: XCTestCase
+
+@end
+
+namespace Test {
+
+class Deck {
+public:
+    using Cards = std::vector<std::reference_wrapper<Card<Deck>>>;
+    
+private:
+    Cards _cards;
+    
+public:
+    using size_type = decltype(_cards)::size_type;
+    
+    Deck(const Cards& cards): _cards{cards} { }
+    Deck(const Deck&) = default;
+    
+    auto draw() -> Card<Deck>& {
+        auto card = this->_cards.front();
+        this->_cards.erase(this->_cards.begin());
+        return card;
+    };
+    auto empty() const -> bool { return this->_cards.empty(); }
+    auto size() const -> size_type { return this->_cards.size(); }
+};
+
+using Card = Card<Deck>;
+using EngCard = EngCard<Deck>;
+using QACard = QACard<Deck>;
+using PMCard = PMCard<Deck>;
+using Player = Player<Deck>;
+
+#pragma mark - Card definitions
```

Comment:
Wow................................... cute :)

Reply:
In the original 61A version they have the course staff with similar cute epithets 😜